File: PST/FUEL Source Code/FUEL/frmMechChecks.cs
// Decompiled with JetBrains decompiler
// Type: FUEL.frmMechChecks
// Assembly: FUEL, Version=2.4.8.0, Culture=neutral, PublicKeyToken=null
// MVID: A3D630D0-0620-425D-8CFB-B67B2AC75507
// Assembly location: C:\hp\FUEL\FUEL.dll

using DevComponents.DotNetBar;
using DevComponents.DotNetBar.Metro;
using DevComponents.DotNetBar.Metro.ColorTables;
using Microsoft.VisualBasic.CompilerServices;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Diagnostics;
using System.Drawing;
using System.Runtime.CompilerServices;
using System.Windows.Forms;

namespace FUEL
{
  [DesignerGenerated]
  public class frmMechChecks : MetroAppForm
  {
    private static List<WeakReference> __ENCList = new List<WeakReference>();
    private IContainer components;
    [AccessedThroughProperty("MetroShell1")]
    private MetroShell _MetroShell1;
    [AccessedThroughProperty("MetroTabPanel1")]
    private MetroTabPanel _MetroTabPanel1;
    [AccessedThroughProperty("MetroTabItem1")]
    private MetroTabItem _MetroTabItem1;
    [AccessedThroughProperty("StyleManager1")]
    private StyleManager _StyleManager1;
    private ctrlMechChecks _ctrlMechChecks;
    private List<PST.PrinterMechChecks> _CheckList;

    [DebuggerNonUserCode]
    static frmMechChecks()
    {
    }

    [DebuggerNonUserCode]
    private static void __ENCAddToList(object value)
    {
      lock (frmMechChecks.__ENCList)
      {
        if (frmMechChecks.__ENCList.Count == frmMechChecks.__ENCList.Capacity)
        {
          int index1 = 0;
          int num = checked (frmMechChecks.__ENCList.Count - 1);
          int index2 = 0;
          while (index2 <= num)
          {
            if (frmMechChecks.__ENCList[index2].IsAlive)
            {
              if (index2 != index1)
                frmMechChecks.__ENCList[index1] = frmMechChecks.__ENCList[index2];
              checked { ++index1; }
            }
            checked { ++index2; }
          }
          frmMechChecks.__ENCList.RemoveRange(index1, checked (frmMechChecks.__ENCList.Count - index1));
          frmMechChecks.__ENCList.Capacity = frmMechChecks.__ENCList.Count;
        }
        frmMechChecks.__ENCList.Add(new WeakReference(RuntimeHelpers.GetObjectValue(value)));
      }
    }

    [DebuggerNonUserCode]
    protected override void Dispose(bool disposing)
    {
      try
      {
        if (!disposing || this.components == null)
          return;
        this.components.Dispose();
      }
      finally
      {
        base.Dispose(disposing);
      }
    }

    [DebuggerStepThrough]
    private void InitializeComponent()
    {
      this.components = (IContainer) new System.ComponentModel.Container();
      this.MetroShell1 = new MetroShell();
      this.MetroTabPanel1 = new MetroTabPanel();
      this.MetroTabItem1 = new MetroTabItem();
      this.StyleManager1 = new StyleManager(this.components);
      this.MetroShell1.SuspendLayout();
      this.SuspendLayout();
      this.MetroShell1.BackColor = Color.White;
      this.MetroShell1.BackgroundStyle.CornerType = eCornerType.Square;
      this.MetroShell1.CanCustomize = false;
      this.MetroShell1.CaptionVisible = true;
      this.MetroShell1.Controls.Add((Control) this.MetroTabPanel1);
      this.MetroShell1.Dock = DockStyle.Top;
      this.MetroShell1.ForeColor = Color.Black;
      this.MetroShell1.HelpButtonText = (string) null;
      this.MetroShell1.HelpButtonVisible = false;
      this.MetroShell1.Items.AddRange(new BaseItem[1]
      {
        (BaseItem) this.MetroTabItem1
      });
      this.MetroShell1.KeyTipsFont = new Font("Tahoma", 7f);
      MetroShell metroShell1_1 = this.MetroShell1;
      Point point1 = new Point(0, 1);
      Point point2 = point1;
      metroShell1_1.Location = point2;
      this.MetroShell1.Name = "MetroShell1";
      this.MetroShell1.SettingsButtonVisible = false;
      MetroShell metroShell1_2 = this.MetroShell1;
      Size size1 = new Size(769, 401);
      Size size2 = size1;
      metroShell1_2.Size = size2;
      this.MetroShell1.SystemText.MaximizeRibbonText = "&Maximize the Ribbon";
      this.MetroShell1.SystemText.MinimizeRibbonText = "Mi&nimize the Ribbon";
      this.MetroShell1.SystemText.QatAddItemText = "&Add to Quick Access Toolbar";
      this.MetroShell1.SystemText.QatCustomizeMenuLabel = "<b>Customize Quick Access Toolbar</b>";
      this.MetroShell1.SystemText.QatCustomizeText = "&Customize Quick Access Toolbar...";
      this.MetroShell1.SystemText.QatDialogAddButton = "&Add >>";
      this.MetroShell1.SystemText.QatDialogCancelButton = "Cancel";
      this.MetroShell1.SystemText.QatDialogCaption = "Customize Quick Access Toolbar";
      this.MetroShell1.SystemText.QatDialogCategoriesLabel = "&Choose commands from:";
      this.MetroShell1.SystemText.QatDialogOkButton = "OK";
      this.MetroShell1.SystemText.QatDialogPlacementCheckbox = "&Place Quick Access Toolbar below the Ribbon";
      this.MetroShell1.SystemText.QatDialogRemoveButton = "&Remove";
      this.MetroShell1.SystemText.QatPlaceAboveRibbonText = "&Place Quick Access Toolbar above the Ribbon";
      this.MetroShell1.SystemText.QatPlaceBelowRibbonText = "&Place Quick Access Toolbar below the Ribbon";
      this.MetroShell1.SystemText.QatRemoveItemText = "&Remove from Quick Access Toolbar";
      this.MetroShell1.TabIndex = 0;
      this.MetroShell1.TabStripFont = new Font("Segoe UI", 10.25f, FontStyle.Bold, GraphicsUnit.Point, (byte) 0);
      this.MetroShell1.Text = "MetroShell1";
      this.MetroTabPanel1.ColorSchemeStyle = eDotNetBarStyle.StyleManagerControlled;
      this.MetroTabPanel1.Dock = DockStyle.Fill;
      MetroTabPanel metroTabPanel1_1 = this.MetroTabPanel1;
      point1 = new Point(0, 51);
      Point point3 = point1;
      metroTabPanel1_1.Location = point3;
      this.MetroTabPanel1.Name = "MetroTabPanel1";
      this.MetroTabPanel1.Padding = new System.Windows.Forms.Padding(3, 0, 3, 3);
      MetroTabPanel metroTabPanel1_2 = this.MetroTabPanel1;
      size1 = new Size(769, 350);
      Size size3 = size1;
      metroTabPanel1_2.Size = size3;
      this.MetroTabPanel1.Style.CornerType = eCornerType.Square;
      this.MetroTabPanel1.StyleMouseDown.CornerType = eCornerType.Square;
      this.MetroTabPanel1.StyleMouseOver.CornerType = eCornerType.Square;
      this.MetroTabPanel1.TabIndex = 1;
      this.MetroTabItem1.Checked = true;
      this.MetroTabItem1.Name = "MetroTabItem1";
      this.MetroTabItem1.Panel = this.MetroTabPanel1;
      this.MetroTabItem1.Text = "Mech Checks";
      this.StyleManager1.ManagerStyle = eStyle.Metro;
      this.StyleManager1.MetroColorParameters = new MetroColorGeneratorParameters(Color.White, Color.FromArgb((int) byte.MaxValue, 163, 26));
      this.AutoScaleDimensions = new SizeF(6f, 13f);
      this.AutoScaleMode = AutoScaleMode.Font;
      size1 = new Size(770, 406);
      this.ClientSize = size1;
      this.Controls.Add((Control) this.MetroShell1);
      this.Name = nameof (frmMechChecks);
      this.MetroShell1.ResumeLayout(false);
      this.MetroShell1.PerformLayout();
      this.ResumeLayout(false);
    }

    internal virtual MetroShell MetroShell1
    {
      [DebuggerNonUserCode] get => this._MetroShell1;
      [DebuggerNonUserCode, MethodImpl(MethodImplOptions.Synchronized)] set => this._MetroShell1 = value;
    }

    internal virtual MetroTabPanel MetroTabPanel1
    {
      [DebuggerNonUserCode] get => this._MetroTabPanel1;
      [DebuggerNonUserCode, MethodImpl(MethodImplOptions.Synchronized)] set => this._MetroTabPanel1 = value;
    }

    internal virtual MetroTabItem MetroTabItem1
    {
      [DebuggerNonUserCode] get => this._MetroTabItem1;
      [DebuggerNonUserCode, MethodImpl(MethodImplOptions.Synchronized)] set => this._MetroTabItem1 = value;
    }

    internal virtual StyleManager StyleManager1
    {
      [DebuggerNonUserCode] get => this._StyleManager1;
      [DebuggerNonUserCode, MethodImpl(MethodImplOptions.Synchronized)] set => this._StyleManager1 = value;
    }

    public frmMechChecks(List<PST.PrinterMechChecks> CheckList)
    {
      this.Load += new EventHandler(this.frmMechChecks_Load);
      frmMechChecks.__ENCAddToList((object) this);
      this.InitializeComponent();
      this._CheckList = CheckList;
    }

    private void frmMechChecks_Load(object sender, EventArgs e)
    {
      this._ctrlMechChecks = new ctrlMechChecks(this._CheckList);
      this.MetroTabPanel1.Controls.Add((Control) this._ctrlMechChecks);
      this._ctrlMechChecks.Dock = DockStyle.Fill;
    }
  }
}

File: PST/FUEL/src/FUEL/frmMechChecks.vb
Imports DevComponents.DotNetBar
Imports DevComponents.DotNetBar.Metro
Imports DevComponents.DotNetBar.Metro.ColorTables
Imports Microsoft.VisualBasic.CompilerServices
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Diagnostics
Imports System.Drawing
Imports System.Runtime.CompilerServices
Imports System.Windows.Forms

Namespace FUEL
    <DesignerGenerated> _
    Public Class frmMechChecks
        Inherits MetroAppForm
        ' Methods
        Public Sub New(ByVal CheckList As List(Of PrinterMechChecks))
            AddHandler MyBase.Load, New EventHandler(AddressOf Me.frmMechChecks_Load)
            frmMechChecks.__ENCAddToList(Me)
            Me.InitializeComponent
            Me._CheckList = CheckList
        End Sub

        <DebuggerNonUserCode> _
        Private Shared Sub __ENCAddToList(ByVal value As Object)
            SyncLock frmMechChecks.__ENCList
                If (frmMechChecks.__ENCList.Count = frmMechChecks.__ENCList.Capacity) Then
                    Dim index As Integer = 0
                    Dim num3 As Integer = (frmMechChecks.__ENCList.Count - 1)
                    Dim num2 As Integer = 0
                    Do While True
                        Dim num4 As Integer = num3
                        If (num2 > num4) Then
                            frmMechChecks.__ENCList.RemoveRange(index, (frmMechChecks.__ENCList.Count - index))
                            frmMechChecks.__ENCList.Capacity = frmMechChecks.__ENCList.Count
                            Exit Do
                        End If
                        Dim reference As WeakReference = frmMechChecks.__ENCList(num2)
                        If reference.IsAlive Then
                            If (num2 <> index) Then
                                frmMechChecks.__ENCList(index) = frmMechChecks.__ENCList(num2)
                            End If
                            index += 1
                        End If
                        num2 += 1
                    Loop
                End If
                frmMechChecks.__ENCList.Add(New WeakReference(value))
            End SyncLock
        End Sub

        <DebuggerNonUserCode> _
        Protected Overrides Sub Dispose(ByVal disposing As Boolean)
            Try 
                If (If((Not disposing OrElse (Me.components Is Nothing)), 0, 1) <> 0) Then
                    Me.components.Dispose
                End If
            Finally
                MyBase.Dispose(disposing)
            End Try
        End Sub

        Private Sub frmMechChecks_Load(ByVal sender As Object, ByVal e As EventArgs)
            Me._ctrlMechChecks = New ctrlMechChecks(Me._CheckList)
            Me.MetroTabPanel1.Controls.Add(Me._ctrlMechChecks)
            Me._ctrlMechChecks.Dock = DockStyle.Fill
        End Sub

        <DebuggerStepThrough> _
        Private Sub InitializeComponent()
            Me.components = New Container
            Me.MetroShell1 = New MetroShell
            Me.MetroTabPanel1 = New MetroTabPanel
            Me.MetroTabItem1 = New MetroTabItem
            Me.StyleManager1 = New StyleManager(Me.components)
            Me.MetroShell1.SuspendLayout
            Me.SuspendLayout
            Me.MetroShell1.BackColor = Color.White
            Me.MetroShell1.BackgroundStyle.CornerType = eCornerType.Square
            Me.MetroShell1.CanCustomize = False
            Me.MetroShell1.CaptionVisible = True
            Me.MetroShell1.Controls.Add(Me.MetroTabPanel1)
            Me.MetroShell1.Dock = DockStyle.Top
            Me.MetroShell1.ForeColor = Color.Black
            Me.MetroShell1.HelpButtonText = Nothing
            Me.MetroShell1.HelpButtonVisible = False
            Dim items As BaseItem() = New BaseItem() { Me.MetroTabItem1 }
            Me.MetroShell1.Items.AddRange(items)
            Me.MetroShell1.KeyTipsFont = New Font("Tahoma", 7!)
            Dim point2 As New Point(0, 1)
            Me.MetroShell1.Location = point2
            Me.MetroShell1.Name = "MetroShell1"
            Me.MetroShell1.SettingsButtonVisible = False
            Dim size2 As New Size(&H301, &H191)
            Me.MetroShell1.Size = size2
            Me.MetroShell1.SystemText.MaximizeRibbonText = "&Maximize the Ribbon"
            Me.MetroShell1.SystemText.MinimizeRibbonText = "Mi&nimize the Ribbon"
            Me.MetroShell1.SystemText.QatAddItemText = "&Add to Quick Access Toolbar"
            Me.MetroShell1.SystemText.QatCustomizeMenuLabel = "<b>Customize Quick Access Toolbar</b>"
            Me.MetroShell1.SystemText.QatCustomizeText = "&Customize Quick Access Toolbar..."
            Me.MetroShell1.SystemText.QatDialogAddButton = "&Add >>"
            Me.MetroShell1.SystemText.QatDialogCancelButton = "Cancel"
            Me.MetroShell1.SystemText.QatDialogCaption = "Customize Quick Access Toolbar"
            Me.MetroShell1.SystemText.QatDialogCategoriesLabel = "&Choose commands from:"
            Me.MetroShell1.SystemText.QatDialogOkButton = "OK"
            Me.MetroShell1.SystemText.QatDialogPlacementCheckbox = "&Place Quick Access Toolbar below the Ribbon"
            Me.MetroShell1.SystemText.QatDialogRemoveButton = "&Remove"
            Me.MetroShell1.SystemText.QatPlaceAboveRibbonText = "&Place Quick Access Toolbar above the Ribbon"
            Me.MetroShell1.SystemText.QatPlaceBelowRibbonText = "&Place Quick Access Toolbar below the Ribbon"
            Me.MetroShell1.SystemText.QatRemoveItemText = "&Remove from Quick Access Toolbar"
            Me.MetroShell1.TabIndex = 0
            Me.MetroShell1.TabStripFont = New Font("Segoe UI", 10.25!, FontStyle.Bold, GraphicsUnit.Point, 0)
            Me.MetroShell1.Text = "MetroShell1"
            Me.MetroTabPanel1.ColorSchemeStyle = eDotNetBarStyle.StyleManagerControlled
            Me.MetroTabPanel1.Dock = DockStyle.Fill
            point2 = New Point(0, &H33)
            Me.MetroTabPanel1.Location = point2
            Me.MetroTabPanel1.Name = "MetroTabPanel1"
            Dim padding2 As New Padding(3, 0, 3, 3)
            Me.MetroTabPanel1.Padding = padding2
            size2 = New Size(&H301, 350)
            Me.MetroTabPanel1.Size = size2
            Me.MetroTabPanel1.Style.CornerType = eCornerType.Square
            Me.MetroTabPanel1.StyleMouseDown.CornerType = eCornerType.Square
            Me.MetroTabPanel1.StyleMouseOver.CornerType = eCornerType.Square
            Me.MetroTabPanel1.TabIndex = 1
            Me.MetroTabItem1.Checked = True
            Me.MetroTabItem1.Name = "MetroTabItem1"
            Me.MetroTabItem1.Panel = Me.MetroTabPanel1
            Me.MetroTabItem1.Text = "Mech Checks"
            Me.StyleManager1.ManagerStyle = eStyle.Metro
            Dim parameters2 As New MetroColorGeneratorParameters(Color.White, Color.FromArgb(&HFF, &HA3, &H1A))
            Me.StyleManager1.MetroColorParameters = parameters2
            Dim ef2 As New SizeF(6!, 13!)
            Me.AutoScaleDimensions = ef2
            Me.AutoScaleMode = AutoScaleMode.Font
            size2 = New Size(770, &H196)
            Me.ClientSize = size2
            Me.Controls.Add(Me.MetroShell1)
            Me.Name = "frmMechChecks"
            Me.MetroShell1.ResumeLayout(False)
            Me.MetroShell1.PerformLayout
            Me.ResumeLayout(False)
        End Sub


        ' Properties
        Friend Overridable Property MetroShell1 As MetroShell
            <DebuggerNonUserCode> _
            Get
                Return Me._MetroShell1
            End Get
            <MethodImpl(MethodImplOptions.Synchronized), DebuggerNonUserCode> _
            Set(ByVal WithEventsValue As MetroShell)
                Me._MetroShell1 = WithEventsValue
            End Set
        End Property

        Friend Overridable Property MetroTabPanel1 As MetroTabPanel
            <DebuggerNonUserCode> _
            Get
                Return Me._MetroTabPanel1
            End Get
            <MethodImpl(MethodImplOptions.Synchronized), DebuggerNonUserCode> _
            Set(ByVal WithEventsValue As MetroTabPanel)
                Me._MetroTabPanel1 = WithEventsValue
            End Set
        End Property

        Friend Overridable Property MetroTabItem1 As MetroTabItem
            <DebuggerNonUserCode> _
            Get
                Return Me._MetroTabItem1
            End Get
            <MethodImpl(MethodImplOptions.Synchronized), DebuggerNonUserCode> _
            Set(ByVal WithEventsValue As MetroTabItem)
                Me._MetroTabItem1 = WithEventsValue
            End Set
        End Property

        Friend Overridable Property StyleManager1 As StyleManager
            <DebuggerNonUserCode> _
            Get
                Return Me._StyleManager1
            End Get
            <MethodImpl(MethodImplOptions.Synchronized), DebuggerNonUserCode> _
            Set(ByVal WithEventsValue As StyleManager)
                Me._StyleManager1 = WithEventsValue
            End Set
        End Property


        ' Fields
        Private Shared __ENCList As List(Of WeakReference) = New List(Of WeakReference)
        Private components As IContainer
        <AccessedThroughProperty("MetroShell1")> _
        Private _MetroShell1 As MetroShell
        <AccessedThroughProperty("MetroTabPanel1")> _
        Private _MetroTabPanel1 As MetroTabPanel
        <AccessedThroughProperty("MetroTabItem1")> _
        Private _MetroTabItem1 As MetroTabItem
        <AccessedThroughProperty("StyleManager1")> _
        Private _StyleManager1 As StyleManager
        Private _ctrlMechChecks As ctrlMechChecks
        Private _CheckList As List(Of PrinterMechChecks)
    End Class
End Namespace

